File: Hafta2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[])
{
    if(argc == 2)
    {
         int key = atoi(argv[1]);
         
       if(!isdigit(key)) 
       {
        string text = get_string("Plaintext:");
        printf("Ciphertext:");
        for(int i = 0; i < strlen(text); i++)
         {
            if(isupper(text[i]))               //text[i] >= 'A' && text[i] <= 'Z'
            {
                printf("%c", (((text[i] - 'A') + key) % 26) + 'A');
            }

            else if( islower(text[i]))      //text[i] >= 'a' && text[i] <= 'z'      
            {
                printf("%c", (((text[i] - 'a') + key) % 26) + 'a');
            }

            else
            {
                printf("%c", text[i]);
            }
           
        }
        printf("\n");
        return 0;
        }
        else
        {
            printf("Usage: ./caesar key\n");
        return 1;
        }
    }

     else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    
    return 0;
}

File: Hafta6/caesar.py
from sys import argv
from cs50 import get_string
import sys

if(len(sys.argv)==2):
    if(str.isdigit(sys.argv[1])):
        key=int(sys.argv[1])
        text=get_string("plaintext: ")
        print("ciphertext: ",end="")
        
        for i in range(len(text)):
            if(str.isupper(text[i])):
                print(f"{chr(((ord(text[i]) - 65 + key) % 26 + 65))}",end="") 
                
            elif(str.islower(text[i])):
                print(f"{chr(((ord(text[i]) - 97 + key) % 26 + 97))}",end="")
                
            else:
                print(text[i],end="")
                
                
               
        print("")
        
        
    else:
        print("Usage: python caesar.py key")
        sys.exit(1)
 
    
else:
    print("Usage: python caesar.py key")
    sys.exit(1)
    
    


























  
    

